catalog/core/catalog-core-downloadaction/src/main/java/org/codice/ddf/catalog/download/action/ResourceDownloadActionEndpoint.java

```diff
@@ -0,0 +1,136 @@
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.download.action;
+
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+
+import javax.management.JMX;
+import javax.management.MBeanException;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+import org.codice.ddf.catalog.resource.download.DownloadToLocalSiteException;
+import org.codice.ddf.catalog.resource.download.ResourceDownloadMBean;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.jknack.handlebars.Handlebars;
+import com.github.jknack.handlebars.Template;
+
+/**
+ * REST endpoint class used to manage metacard product downloads.
+ */
+@Path("/")
+public class ResourceDownloadActionEndpoint {
+    
+    private static final Logger LOGGER = LoggerFactory.getLogger(ResourceDownloadActionEndpoint.class);
+    
+    public static final String CONTEXT_PATH = "/internal/catalog/download/cache";
+
+    public static final String SOURCE_PARAM = "source";
+
+    public static final String METACARD_PARAM = "metacard";
+
+    private static final String RESPONSE_HTML_TEMPLATE = "response";
+
+    private static final String SUCCESS_MESSAGE = "Download of resource started successfully";
+
+    private static final String FAILURE_MESSAGE = "Failed to start download of resource";
+
+    private final ResourceDownloadMBean resourceDownloadMBean;
+
+    private final Template template;
+
+    public ResourceDownloadActionEndpoint(Handlebars handlebars) {
+        this.resourceDownloadMBean = createResourceDownloadMBeanProxy();
+        try {
+            template = handlebars.compile(RESPONSE_HTML_TEMPLATE);
+        } catch (IOException e) {
+            String message = String.format("Unable to compile handlebars template [%s].",
+                    RESPONSE_HTML_TEMPLATE);
+            LOGGER.debug(message);
+            throw new RuntimeException(message, e);
+        }
+    }
+    
+    /**
+     * Starts an asynchronous download of a specific metacard resource to the local site.
+     *
+     * @param sourceId   ID of the federated source where the resource should be downloaded from
+     * @param metacardId ID of the metacard that contains the resource to download
+     * @return response object containing successful start of download message
+     * @throws DownloadToCacheOnlyException
+     */
+    @GET
+    public Response copyToLocalSite(@FormParam(SOURCE_PARAM) String sourceId,
```

Comment:
Instead of returning Response, you could return an object like String when successful or an exception if not.  Then a message body writer and exception mapper could generate HTML using `generateHtmlResponse`.